components/features/dashboard/SchoolSwitcher.tsx

```diff
@@ -56,7 +54,7 @@ export function SchoolSwitcher({ activeSchoolId }: Props) {
     );
   }
 
-  if (!schools) return null;
+  if (!schools?.joined) return null;
```

Comment:
This won't work because arrays are always truthy.
```suggestion
  if (schools?.joined.length == 0) return null;
```